module/data/item/facility.mjs

```diff
@@ -208,6 +208,9 @@ export default class FacilityData extends ItemDataModel.mixin(ActivitiesTemplate
 
     if ( this.disabled ) this.progress.order = "repair";
 
+    // Trade
+    if ( Number.isFinite(this.trade.creatures.max) ) this.trade.creatures.max -= this.trade.pending.creatures.length;
```

Comment:
I think this is hitting into a double-preparation bug in V12, ending up with the max creatures being negative whenever a trade is in progress and preventing any creatures from actually being added when it is complete.

```suggestion
    if ( Number.isFinite(this.trade.creatures.max) ) {
      this.trade.creatures.max = this._source.trade.creatures.max -= this.trade.pending.creatures.length;
    }
```